File: recycler-adapter/src/main/java/com/android/oleksandrpriadko/recycler_adapter/BaseHolderPicasso.java
package com.android.oleksandrpriadko.recycler_adapter;

import android.support.annotation.DrawableRes;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;

import com.android.oleksandrpriadko.recycler_adapter.R;
import com.squareup.picasso.Picasso;

public class BaseHolderPicasso extends RecyclerView.ViewHolder {
    public BaseHolderPicasso(View itemView) {
        super(itemView);
    }

    protected void loadImage(String url,
                             ImageView imageView,
                             @DrawableRes int placeHolderId,
                             @DrawableRes int errorID) {
        getPicasso()
                .load(url)
                .placeholder(placeHolderId)
                .error(errorID)
                .into(imageView);
    }

    protected void loadImage(String url,
                             ImageView imageView,
                             @DrawableRes int placeHolderId) {
        getPicasso()
                .load(url)
                .placeholder(placeHolderId)
                .error(R.drawable.ic_error_outline_black_24dp)
                .into(imageView);
    }

    private Picasso getPicasso() {
        Picasso picasso = Picasso.with(itemView.getContext());
        picasso.setIndicatorsEnabled(false);
        picasso.setLoggingEnabled(false);
        return picasso;
    }
}

File: recycler-adapter/src/main/java/com/android/oleksandrpriadko/recycler_adapter/BaseHolderPicasso.kt
package com.android.oleksandrpriadko.recycler_adapter

import android.view.View
import android.widget.ImageView
import androidx.annotation.DrawableRes
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso

open class BaseHolderPicasso(itemView: View) : RecyclerView.ViewHolder(itemView) {

    private val picasso: Picasso
        get() {
            val picasso = Picasso.with(itemView.context)
            picasso.setIndicatorsEnabled(false)
            picasso.isLoggingEnabled = false
            return picasso
        }

    protected fun loadImage(url: String,
                            imageView: ImageView,
                            @DrawableRes placeHolderId: Int,
                            @DrawableRes errorID: Int) {
        picasso
                .load(url)
                .placeholder(placeHolderId)
                .error(errorID)
                .into(imageView)
    }

    protected fun loadImage(url: String?,
                            imageView: ImageView,
                            @DrawableRes placeHolderId: Int) {
        picasso
                .load(url)
                .placeholder(placeHolderId)
                .error(R.drawable.ic_error_outline_black_24dp)
                .into(imageView)
    }
}
